### PR TITLE
Upgrade Slate, and have successful platform doc builds…

### PR DESCRIPTION
**Pull request by fhanik**
…  on both Mac and Linux using Ruby 3.3.5

Step 1 - Upgrade dependencies of Slate ruby gems

Note: May require pipeline modifications to include Ruby 3.3.x

### CODE REVIEW

**Comment by strehle** [comment]
hI @fhanik ! back in UAA ?
So your PR is welcome, but we need to update the runtime in the voters 
, e.g.
error from there

Your Ruby version is 3.0.7, but your Gemfile specified >= 3.3

FYI @peterhaochen47 / @hsinn0

File: uaa/slate/Gemfile [hunk 1]
```diff
@@ -1,15 +1,17 @@
-ruby '>= 3.0'
+ruby '>= 3.3'
 source 'https://rubygems.org'
 
 gem 'bundler', '~> 2.2'
```
Comment (strehle) [inline uaa/slate/Gemfile:4]:
```suggestion
gem 'bundler', '~> 2.5'
```

File: uaa/slate/Gemfile [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 gem 'middleman-sprockets', '~> 4.1'
 gem 'rouge', '~> 3.21'
 gem 'redcarpet', '~> 3.6.0'
 gem 'nokogiri', '~> 1.16.5'
 gem 'sass'
 gem 'webrick'
-gem 'mini_racer', '~> 0.4.0', :platform => :ruby
+gem 'opal'
+gem 'opal-sprockets'
+
```
Comment (strehle) [inline uaa/slate/Gemfile:17]:
```suggestion
gem 'mini_racer', '~> 0.16.0', :platform => :ruby
```

File: uaa/slate/Gemfile.lock [hunk 3]
```diff
@@ -137,16 +142,17 @@ DEPENDENCIES
...
   rouge (~> 3.21)
   sass
   webrick
 
 RUBY VERSION
-   ruby 3.0
+   ruby 3.3.5
 
 BUNDLED WITH
    2.2.22
```
Comment (strehle) [inline uaa/slate/Gemfile.lock:158]:
```suggestion
   2.5.21
```

File: uaa/slate/Gemfile.lock
Comment (strehle) [inline uaa/slate/Gemfile.lock:1]: DEPENDENCIES
  bundler (~> 2.5)

**Review by strehle — COMMENTED** [review]
without mini_racer I see error

bundler: failed to load command: middleman (/usr/local/bundle/bin/middleman)
/usr/local/bundle/gems/execjs-2.9.1/lib/execjs/runtimes.rb:68:in `autodetect': Could not find a JavaScript runtime. See https://github.com/rails/execjs for a list of available runtimes. (ExecJS::RuntimeUnavailable)
        from /usr/local/bundle/gems/execjs-2.9.1/lib/execjs.rb:5:in `<module:ExecJS>'

**Comment by strehle** [comment]
> Note: May require pipeline modifications to include Ruby 3.3.x

docker images was updated, e.g. you can test with 

docker run --privileged --tty --interactive --shm-size=1 -v.:/root/uaa cfidentity/uaa-generate-docs /root/uaa/scripts/generate-docs.sh

**Comment by fhanik** [comment]
> > Note: May require pipeline modifications to include Ruby 3.3.x
> 
> docker images was updated, e.g. you can test with
> 
> docker run --privileged --tty --interactive --shm-size=1 -v.:/root/uaa cfidentity/uaa-generate-docs /root/uaa/scripts/generate-docs.sh

Awesome. I suspect this is the build image too. I will resolve it. 

Am I back…not quite yet, but it may be a possibility in the future. @strehle

**Comment by hsinn0** [comment]
The docker images update broke a main pipeline job https://bosh.ci.cloudfoundry.org/teams/uaa/pipelines/uaa-acceptance-gcp/jobs/test-generate-api-docs/builds/316. As I do not have resource to put an effort on resolving it at the moment, I might have to revert the docker images update to make the pipeline green. Will see.

**Comment by hsinn0** [comment]
> The docker images update broke a main pipeline job https://bosh.ci.cloudfoundry.org/teams/uaa/pipelines/uaa-acceptance-gcp/jobs/test-generate-api-docs/builds/316. As I do not have resource to put an effort on resolving it at the moment, I might have to revert the docker images update to make the pipeline green. Will see.

I realize that the main pipeline job failure is likely to be resolved when additional changes for this PR are made. So I will wait for the changes in this PR for now.

**Comment by strehle** [comment]
[like]  Strehle, Markus reacted to your message:
________________________________

**Comment by fhanik** [comment]
@strehle @hsinn0 

Since there was a `scripts/jasmine-tests` I assumed your build images would already have Javascript (nodejs) installed.

The dockerfile (can you point me to it so I can send a PR?) would need to include

```
# install NVM
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.1/install.sh | bash

# load NVM
 export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion

# install npm (I used 18.18.2 cause it works)
nvm install 18.18.2
```

Now `./gradlew generateDocs` will work. There is a small update needed to jasmine test that I'm including in this PR too

**Comment by hsinn0** [comment]
> The dockerfile (can you point me to it so I can send a PR?) would need to include

https://github.com/cloudfoundry/uaa-ci/blob/main/concourse/dockerfiles/uaa-generate-docs/Dockerfile
@fhanik , I just gave you w perm to the repo so u can push a PR.

**Comment by fhanik** [comment]
@hsinn0 @strehle 

I was working on the uaa-ci patch, but lost access while doing so. I no longer can see the repository

here is the patch for the Dockerfile

```
# install NVM for JavaScript/nodejs
RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.1/install.sh | bash

# load NVM into the profile and install npm
RUN \
   export NVM_DIR="$HOME/.nvm" \
    && [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" \
    && [ -s "$NVM_DIR/bash_completion" ] \
    && nvm install 18.18.2
```

add those, and we should be green on `generateDocs` as well on `scripts/jasmine-tests` again

**Comment by strehle** [comment]
Can you @fhanik please sign CLA -  see error . I know you was an approver for UAA already, but maybe you have to re-assign (because of changes) but we must only accept PR with a green CLA...  https://tenor.com/b1jfv.gif

**Comment by strehle** [comment]
> nvm-sh

I like more the OS dependent approach because then we dont have to maintain broker URLs, so we have node installation in uaa-singular and here we do:

```
## Install node
RUN \
  curl -sL https://deb.nodesource.com/setup_20.x | sudo -E bash -
RUN \
    apt-get install -y nodejs && \
    npm install npm@latest -g
```

On my machine I use and like nvm but in a docker we dont need the flexiblity of different node versions

File: scripts/jasmine-tests [hunk 4]
```diff
@@ -3,7 +3,7 @@ set -eu -o pipefail
 
 UAA_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
 
-pushd "${UAA_DIR}"
+pushd "${UAA_DIR}/uaa"
```
Comment (strehle) [inline scripts/jasmine-tests:6]: why ?

**Comment by fhanik** [comment]
> Can you @fhanik please sign CLA - see error . I know you was an approver for UAA already, but maybe you have to re-assign (because of changes) but we must only accept PR with a green CLA... https://tenor.com/b1jfv.gif

I signed one, but it’s probably sitting in an approval queue at Broadcom. 

I’ll see if I can track it down

**Comment by strehle** [comment]
> approval queue at Broadcom.

cool, thanks. 

You may have seen, that voter for docs generation now is green. .e.g. https://github.com/cloudfoundry/uaa/actions/runs/11281815112/job/31408214542?pr=3081

File: scripts/jasmine-tests (hunk 4 above)
Comment (fhanik) [reply scripts/jasmine-tests:6]: That’s the location of the JavaScript tests that the script is trying to run, isn’t it?

Try to run the old ‘jasmine-tests’ script, it fails for me, due to the wrong location.

**Comment by hsinn0** [comment]
> I was working on the uaa-ci patch, but lost access while doing so. I no longer can see the repository

@fhanik, yeah, the cloudfoundry automation overwrote the access settings. I see that @strehle made the change in `uaa-ci` though, so we are good there.

**Comment by fhanik** [comment]
I just learned that if I'm a member of the `pivotal-cf` org, I should not need a CLA. Let me close this PR and reopen to trigger another check.

**Pull request closed without merging**